geti_sdk/deployment/predictions_postprocessing/results_converter/results_to_prediction_converter.py

```diff
@@ -355,29 +424,45 @@ class SegmentationToPredictionConverter(InferenceResultsToPredictionConverter):
     :param label_schema: LabelSchema containing the label info of the task
     """
 
-    domain = Domain.SEGMENTATION
-
     def __init__(self, label_schema: LabelSchema):
         self.labels = label_schema.get_labels(include_empty=False)
         # NB: index=0 is reserved for the background label
         self.label_map = dict(enumerate(self.labels, 1))
 
     def convert_to_prediction(
-        self, predictions: ImageResultWithSoftPrediction, **kwargs  # noqa: ARG002
+        self, inference_results: ImageResultWithSoftPrediction, **kwargs  # noqa: ARG002
     ) -> Prediction:
         """
-        Convert ModelAPI instance segmentation predictions to Prediction object.
+        Convert ModelAPI instance segmentation inference results to Prediction object.
 
-        :param predictions: segmentation represented in ModelAPI format
+        :param inference_results: segmentation represented in ModelAPI format
         :return: Prediction object containing the contour polygon obtained from the segmentation
         """
         annotations = create_annotation_from_segmentation_map(
-            hard_prediction=predictions.resultImage,
-            soft_prediction=predictions.soft_prediction,
+            hard_prediction=inference_results.resultImage,
+            soft_prediction=inference_results.soft_prediction,
             label_map=self.label_map,
         )
         return Prediction(annotations)
 
+    def convert_saliency_map(
+        self,
+        inference_results: NamedTuple,
+        image_shape: Tuple[int, int, int],
+    ) -> Optional[Dict[str, np.ndarray]]:
+        """
+        Extract a saliency map from inference results and return in a unified format.
+
+        :param inference_results: classification labels represented in ModelAPI format (label_index, label_name, confidence)
+        :param image_shape: shape of the input image
+        :return: Prediction object with corresponding label
+        """
+        saliency_map = inference_results.saliency_map
+        if len(saliency_map) == 0:
+            return None
+        saliency_map = np.transpose(saliency_map, (2, 0, 1))  # shape: (N classes, h, w)
+        return {label.name: saliency_map[i + 1] for i, label in enumerate(self.labels)}
```

Comment:
Isn't saliency map zero-indexed as well? i.e. shouldn't it just be ` saliency_map[i]` ? Or does the ` saliency_map[0]`  entry contain the saliency map for the background?

Reply:
I tried to find out how it is processed in model API, but it is not clear at all.
Empirically, yes, this is the correct way to index the map.